File: backend/src/models/Room.js
const mongoose = require("mongoose");

const roomSchema = new mongoose.Schema({
    roomNumber: { type: String, required: true, unique: true },
    name: { type: String, required: true },
    type: { type: String, required: true, enum: ["STANDARD", "DELUXE", "SUITE", "PRESIDENTIAL"] },
    price: { type: Number, required: true },
    desc: { type: String },
    image: { type: String },
    isAvailable: { type: Boolean, default: true },
});

module.exports = mongoose.model("Room", roomSchema);

File: backend-java/src/main/java/com/oceanview/reservation/models/Room.java
package com.oceanview.reservation.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.Data;

@Data
@Document(collection = "rooms")
public class Room {
    @Id
    private String id;
    private String roomNumber;
    private String name;
    private String type; // "STANDARD", "DELUXE", "SUITE", "PRESIDENTIAL"
    private Double price;
    private String desc;
    private String image;
    private Boolean isAvailable = true;
}
